sklearn/model_selection/_search.py

```diff
@@ -965,11 +965,13 @@ def _store(key_name, array, weights=None, splits=False, rank=False):
             results["std_%s" % key_name] = array_stds
 
             if rank:
-                rank_result = rankdata(-array_means, method="min")
                 # when input is nan, scipy >= 1.10 rankdata returns nan. To
                 # keep previous behaviour nans are set to the
-                # maximum possible rank
-                rank_result[np.isnan(rank_result)] = len(rank_result)
+                # minimum value in the array before ranking
+                array_means = np.asarray(
+                    [min(array_means) - 1 if np.isnan(a) else a for a in array_means]
```

Comment:
I would suggest to compute `min(arrays_means)` only once.

Also, because `array_means` is a numpy array, it is faster to use masking or indexing instead of python loops: `array_means[np.isnan(array_means)] = min(array_means) - 1`.